src/tests/ftest/util/command_utils.py

```diff
@@ -176,4 +200,353 @@ def __str__(self):
+        """Create a ExecutableCommand object.
+
+        Uses Avocado's utils.process module to run a command str provided.
+
+        Args:
+            namespace (str): yaml namespace (path to parameters)
+            command (str): string of the command to be executed.
+            path (str, optional): path to location of command binary file.
+                Defaults to "".
+            subprocess (bool, optional): whether the command is run as a
+                subprocess. Defaults to False.
+        """
+        super(ExecutableCommand, self).__init__(namespace, command, path)
+        self._process = None
+        self.run_as_subprocess = subprocess
+        self.timeout = None
+        self.verbose = True
+        self.env = None
+        self.sudo = False
+
+    def run(self):
+        """Run the command.
+
+        Raises:
+            CommandFailure: if there is an error running the command
+
+        """
+        if self.run_as_subprocess:
+            self._run_subprocess()
+        else:
+            self._run_process()
+
+    def _run_process(self):
+        """Run the command as a foreground process.
+
+        Raises:
+            CommandFailure: if there is an error running the command
+
+        """
+        command = self.__str__()
+        kwargs = {
+            "cmd": command,
+            "timeout": self.timeout,
+            "verbose": self.verbose,
+            "allow_output_check": "combined",
+            "shell": True,
+            "env": self.env,
+            "sudo": self.sudo,
+        }
+        try:
+            # Block until the command is complete or times out
+            process.run(**kwargs)
+
+        except process.CmdError as error:
+            # Command failed or possibly timed out
+            msg = "Error occurred running '{}': {}".format(command, error)
+            print(msg)
+            raise CommandFailure(msg)
+
+    def _run_subprocess(self):
+        """Run the command as a sub process.
+
+        Raises:
+            CommandFailure: if there is an error running the command
+
+        """
+        if self._process is None:
+            # Start the job manager command as a subprocess
+            kwargs = {
+                "cmd": self.__str__(),
+                "verbose": self.verbose,
+                "allow_output_check": "combined",
+                "shell": True,
+                "env": self.env,
+                "sudo": self.sudo,
+            }
+            self._process = process.SubProcess(**kwargs)
+            self._process.start()
+
+            # Deterime if the command has launched correctly using its
+            # check_subprocess_status() method.
+            if not self.check_subprocess_status(self._process):
+                msg = "Command '{}' did not launch correctly".format(self)
+                print(msg)
+                raise CommandFailure(msg)
+        else:
+            print("Process is already running")
+
+    def check_subprocess_status(self, subprocess):
```

Comment:
(pylint-unused-argument) Unused argument 'subprocess'
(pylint-no-self-use) Method could be a function